src/librustdoc/lib.rs

```diff
@@ -448,6 +449,9 @@ fn main_args(args: &[String]) -> MainResult {
             early_error(ErrorOutputType::default(), &err.to_string());
         }
     };
+    let sopts = build_session_options(&matches);
```

Comment:
I'm unsure if this is the best way to do it since I think we'll then be calling `build_session_options()` twice, once in rustdoc and once in rustc. But I don't know of a good alternative.

Reply:
You can also move the `rustc_driver::init_env_logger` below the `let options` below and access `options.debugging_opts.always_color_logs`